data-prepper-plugins/translate-processor/src/test/java/org/opensearch/dataprepper/plugins/processor/translate/TranslateProcessorTest.java

```diff
@@ -0,0 +1,270 @@
+package org.opensearch.dataprepper.plugins.processor.translate;
+
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.Mock;
+import org.mockito.junit.jupiter.MockitoExtension;
+import org.opensearch.dataprepper.expression.ExpressionEvaluator;
+import org.opensearch.dataprepper.metrics.PluginMetrics;
+import org.opensearch.dataprepper.model.event.Event;
+import org.opensearch.dataprepper.model.event.JacksonEvent;
+import org.opensearch.dataprepper.model.plugin.InvalidPluginConfigurationException;
+import org.opensearch.dataprepper.model.record.Record;
+
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Collections;
+import java.util.AbstractMap;
+
+import static org.mockito.Mockito.lenient;
+import static org.mockito.Mockito.when;
+
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+import static org.hamcrest.CoreMatchers.is;
+
+@ExtendWith(MockitoExtension.class)
+class TranslateProcessorTest {
+
+    @Mock
+    private PluginMetrics pluginMetrics;
+    @Mock
+    private TranslateProcessorConfig mockConfig;
+
+    @Mock
+    private RegexParameterConfiguration mockRegexConfig;
+
+    @Mock
+    private ExpressionEvaluator expressionEvaluator;
+
+    @BeforeEach
+    void setup(){
+        lenient().when(mockConfig.getSource()).thenReturn("sourceField");
+        lenient().when(mockConfig.getTarget()).thenReturn("targetField");
+    }
+
+    @Test
+    public void test_string_keys_in_map(){
+        when(mockConfig.getMap()).thenReturn(createMapEntries(createMapping("key1","mappedValue1")));
```

Comment:
For every test, add positive and negative test case. For example, for this test, add the following lines at the end of the function
```
record = getEvent("key2");
final List<Record<Event>> translatedRecords = (List<Record<Event>>) processor.doExecute(Collections.singletonList(record));
assertThat(editedRecords.get(0).getData().containsKey("targetField"), is(false));
```

You can also replace `assertThat(...is(false))` with `assertTrue(...)`


```